src/App.tsx

```diff
@@ -3,12 +1,58 @@
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
+import { STATUSES, TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<string>(STATUSES.ALL);
+  const [query, setQuery] = useState<string>('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos().then(res => setTodoList(res));
+  }, []);
+
+  const filteredTodoListByStatus = (): Todo[] => {
+    switch (status) {
+      case STATUSES.COMPLETED:
+        return todoList.filter(todo => todo.completed);
+
+      case STATUSES.ACTIVE:
+        return todoList.filter(todo => !todo.completed);
+
+      case STATUSES.ALL:
+      default:
+        return todoList;
+    }
+  };
```

Comment:
The `filteredTodoListByStatus` function could be optimized by using a single return statement. Instead of using a `switch` statement, consider using a filter method with a condition that checks the `status` variable.